src/calculateRank.js

```diff
@@ -1,12 +1,114 @@
-function exponential_cdf(x) {
-  return 1 - 2 ** -x;
+function searchSorted(arr, x) {
```

Comment:
As the size of the list is small and constant, it may be interesting to use a dictionary instead of a list and a "complex" find algorithm ?

Reply:
I am not sure how you would use a dictionary to find the index `i` for which `arr[i - 1] <= x < arr[i]`.